### DISCUSSION TITLE
Доступ к просмотоу строк в зависимости от пользователя

### DISCUSSION THREAD

**Discussion by Romanufa999** [totumonline/totum-ru-issues-and-discussions#608] [Как это сделать?]
Есть таблица лидов. 
Создал колонку - ответственный, где как то указываю или имя или логин пользователя. 

Как сделать чтобы каждый пользователь видел и правил только те Лиды, у которых его логин указан ответственным? А лидов других пользователей никак не мог просмотреть.

**Comment by erpic-ru** [comment]
Создаете префильтр на это поле, пишете код, ограничиваете доступ к нему по роли и в коде пишете

`=:if(condition $#nr = роль сотрудника; then: $#nu; else: "*ALL*") `
либо обратное условие 
https://ru.docs.totum.online/how-does-access-to-change-and-prefilter-visibility-affect-filtering

**Comment by Romanufa999** [comment]
о, классно, так проктически можно реализовать задумку. в идеале хочу так:
Пользователь - у него поле - метка.
Лид - поле - метка.
Если и у лида метка (более дллинная) содержит метку пользвател - то пользователь видит только этих лидов.
можете прм подсказать как это реализовать - буду очень благодарен

**Reply by erpic-ru** [reply]
Метка-то зачем? Достаточно лид - пользователь и префильтр на пользователях

**Reply by Romanufa999** [reply]
из каналов лидогенераций - лиды поступают с меткми , чтобы атоматом на пользователя цеплять

**Reply by erpic-ru** [reply]
Можно кодом находить и подставлять менеджера по метке и фильтровать по нему, либо префильтр на метку сделать, который будет ее выбирать из таблицы по пользователю. Можно в пользователей колонку добавить, либо отдельную табличку пользователь - метка сделать

**Reply by Romanufa999** [reply]
мне бы готовое чтото  - готовый код как эт реализовать )

**Reply by ttm-alexeypolunin** [reply]
У меня есть таблица соответсвия меток и пользователей. Метка приходит в поле #metka, как мне в поле с пользователем выбрать правильного пользователя по метке?

```
=: select(table: 'metka_user'; field: 'user'; where: 'metka' = #metka)
```

**Reply by Romanufa999** [reply]
как вырбрать не метка   = ,а Поле талицы - содержит метку пользоателя ?

**Comment by Romanufa999** [comment]
https://ru.docs.totum.online/how-does-access-to-change-and-prefilter-visibility-affect-filtering
это классная статья. думаю то что надо.

но можно ли по подробнее пожалуйста @erpic-ru   для чайника - какие поля в таблице с лидами и какие пол в таблице с пользователями конкретнее.
потому что я потыкал - пока не получилось
